### PR TITLE
rafthttp: fix TestStopBlockedPipeline

### LINKED ISSUE #2946: TestStopBlockedPipeline fail
--- FAIL: TestStopBlockedPipeline-2 (1.00s)
	pipeline_test.go:210: failed to stop pipeline in 1s
FAIL
coverage: 81.2% of statements
FAIL	github.com/coreos/etcd/rafthttp	1.213s

### PR DESCRIPTION
**Pull request by yichengq**
- rafthttp: wait 1ms before enabling cancel …
CancelRequest only effects on in-flight request, so we need to wait
for Do(request) called before enabling cancel.

- rafthttp: fix TestStopBlockedPipeline …
Refactor the fake cancel implementation.

The old one may cancel other in-flight message in random, which leaves
the original target message blocked forever.

fixes #2946

### CODE REVIEW

**Comment by yichengq** [comment]
@xiang90 addressed all comments.

File: rafthttp/pipeline.go [hunk 1]
```diff
@@ -150,6 +150,7 @@ func (p *pipeline) post(data []byte) (err error) {
 	}()
 	done := make(chan struct{}, 1)
 	go func() {
+		waitSchedule()
```
Comment (xiang90) [inline rafthttp/pipeline.go]: how about moving this to line 157?

**Comment by xiang90** [comment]
The change cannot compile.

```
gopath/src/github.com/coreos/etcd/rafthttp/pipeline_test.go:246: undefined: cancel
```

File: rafthttp/pipeline.go (hunk 1 above)
Comment (yichengq) [inline rafthttp/pipeline.go]: good idea. moving to line 157 let it skip waiting when the request is done in less than 1ms. will change.

**Comment by xiang90** [comment]
LGTM

**Pull request merged**